internal/provider/provider.go

```diff
@@ -349,7 +352,20 @@ func New(version string) func() provider.Provider {
 	return func() provider.Provider {
 		return &ImageTestProvider{
 			version: version,
-			store:   NewProviderStore(),
 		}
 	}
 }
+
+// mergeResourceSchemas merges the given schemas into a single map. priority is last to
+// first.
+func mergeResourceSchemas(schemas ...map[string]rschema.Attribute) map[string]rschema.Attribute {
+	result := make(map[string]rschema.Attribute)
+
+	for _, s := range schemas {
+		for k, v := range s {
+			result[k] = v
+		}
+	}
+
+	return result
+}
```

Comment:
Any reason not to use the existing implementation?
https://github.com/chainguard-dev/terraform-provider-imagetest/blob/18cc605d93c81d0dfcb9a2ef6313efa5cff44de8/internal/util/map_utils.go#L5-L15

This refactor doesn't use it but doesn't remove it either.

Reply:
the goal was to 🪓 the `utils` pkg, I'll see if that'll squeeze into this pr!